### PR TITLE
replace atoi with strtol

### PR DESCRIPTION
**Pull request by maipbui**
From Semgrep: https://semgrep.dev/r?q=c.lang.correctness.incorrect-use-ato-fn.incorrect-use-ato-fn

> Avoid the 'ato*()' family of functions. Their use can lead to undefined behavior, integer overflows, and lack of appropriate error handling. Instead prefer the 'strtol*()' family of functions.

From atoi() man page: https://www.man7.org/linux/man-pages/man3/atoi.3.html

> The atoi() function converts the initial portion of the string pointed to by nptr to int.  The behavior is the same as
>            strtol(nptr, NULL, 10);
> except that atoi() does not detect errors.

Therefore, replace atoi() with strtol()

Verified by kill dhcpmon process, installing new .deb change to dhcp_relay docker, execute dhcpmon process, check if counters is printing in syslog, compare counters before and after this change.
logs: 
[before.txt](https://github.com/sonic-net/sonic-dhcpmon/files/11377335/before.txt)
[after.txt](https://github.com/sonic-net/sonic-dhcpmon/files/11377336/after.txt)

example commands: 
[command.txt](https://github.com/sonic-net/sonic-dhcpmon/files/11423945/command.txt)

### CODE REVIEW

**Comment by maipbui** [comment]
@qiluo-msft @kellyyeh could you help review this PR? I don't have permission to add reviewers.

File: src/main.cpp [hunk 1]
```diff
@@ -150,15 +150,15 @@ int main(int argc, char **argv)
             i++;
             break;
         case 's':
-            snaplen = atoi(argv[i + 1]);
+            snaplen = strtol(argv[i + 1], NULL, 10);
```
Comment (qiluo-msft) [inline src/main.cpp:153]: >strtol [](http://example.com/codeflow?start=22&length=6)

This new function has better error handling, you may consider add more code to check for error cases, and report back to user.

File: src/main.cpp [hunk 2]
```diff
@@ -116,9 +117,10 @@ int main(int argc, char **argv)
     int i;
     int window_interval = dhcpmon_default_health_check_window;
     int max_unhealthy_count = dhcpmon_default_unhealthy_max_count;
-    size_t snaplen = dhcpmon_default_snaplen;
+    int snaplen = dhcpmon_default_snaplen;
```
Comment (qiluo-msft) [inline src/main.cpp:120]: >int [](http://example.com/codeflow?start=4&length=3)

In theory, it is `long int`.

Comment (qiluo-msft) [reply src/main.cpp]: Sorry for misleading. Original types are good.

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@kellyyeh could you help review?

**Review by jcaiMR — APPROVED** [review]

**Review by kellyyeh — APPROVED** [review]

**Pull request merged**